### PR TITLE
Document offset inclusivity and exclusivity in Protobuf

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
As brought up in https://discuss.daml.com/t/using-offsets/1731, the
inclusivity and exclusivity of start and end offsets as exposed by
the Ledger API is not well documented.

This commit clearly states which offset is inclusive and which is
exclusive on all relevant endpoints (command completion and
transaction services).

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_completion_service.proto [hunk 1]
```diff
@@ -69,6 +69,7 @@ message CompletionStreamRequest {
   repeated string parties = 3;
 
   // This field indicates the minimum offset for completions. This can be used to resume an earlier completion stream.
+  // This offset is exclusive: the response will not contain any command whose offset is less than or equal to this.
```
Comment (cocreature) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_completion_service.proto:72]: I find negation always a bit harder to read. Maybe ‘will only contain commands whose offset is strictly greater to this`?

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto [hunk 2]
```diff
@@ -88,10 +88,12 @@ message GetTransactionsRequest {
   string ledger_id = 1;
 
   // Beginning of the requested ledger section.
+  // This offset is exclusive: the response will not contain any transaction whose offset is less than or equal to this.
   // Required
   LedgerOffset begin = 2;
 
   // End of the requested ledger section.
+  // This offset is inclusive: the response will not contain any transaction whose offset is strictly greater than this.
```
Comment (cocreature) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto:96]: Maybe `will only contain transactions whose offset is less than or equal to this`?

**Review by cocreature — APPROVED** [review]
Thanks! One small suggestion on the wording

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_completion_service.proto (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_completion_service.proto:72]: Ah, thanks, I struggled a bit with this!

Comment (stefanobaghino-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_completion_service.proto]: Addressed by https://github.com/digital-asset/daml/pull/8145/commits/bb8621be35dd50afa01c52909147198a72e84fed

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto (hunk 2 above)
Comment (stefanobaghino-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto]: Addressed by https://github.com/digital-asset/daml/pull/8145/commits/bb8621be35dd50afa01c52909147198a72e84fed

**Review by gerolf-da — APPROVED** [review]

**Pull request merged**